File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.cc
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
#include <set>
#include <ostream>
#include <string>
#include <cstring>
#include <sstream>
#include <time.h>

using namespace std;

int n;

int fib(int n) {
    if (n == 1 || n == 2){
        return 1;
    }
    return fib(n - 1) + fib(n - 2);
}

int main(int argc, const char * argv[]) {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    
    cin >> n;
    
    cout << fib(n) << " " << n - 2 <<  endl;
}

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
fib = [0] * 41
fib[1], fib[2] = 1, 1
n = int(input())
for i in range(3, n + 1):
    fib[i] = fib[i - 1] + fib[i - 2]
print(fib[n], n - 2)
